reqhandler/request_handler.go

```diff
@@ -21,3 +22,34 @@ func readRequest(udpConn net.PacketConn) ([]byte, error) {
 	_, _, err := udpConn.ReadFrom(b)
 	return b, err
 }
+
+func extractID(d []byte) uint16 {
+	return (uint16(d[0]) << 8) | uint16(d[1])
```

Comment:
We should do this check at a higher level. We know that the header length is 12 bytes. If it isn't, we respond at a higher level. But I don't think we need to duplicate this check at a lower level each time.